packages/repository/src/query.ts

```diff
@@ -265,15 +265,7 @@ export class WhereBuilder<MT extends object = AnyObject> {
    * @param clause And/Or/Condition clause
    */
   cast(clause: AndClause<MT> | OrClause<MT> | Condition<MT>): Where<MT> {
-    const w: Where<MT> = {};
-    if ('and' in clause) {
-      w.and = clause.and;
-    } else if ('or' in clause) {
-      w.or = clause.or;
-    } else {
-      Object.assign(w, clause);
-    }
-    return w;
+    return clause;
```

Comment:
Hmm, are you sure this change is preserving current behavior?

A test case to consider:

```ts
cast({and: [{x: 1}, {y: 2}], z: 3})
```

Before your change, cast returns

```
{and: [{x: 1}, {y: 2}]}
```

After your change, cast returns

```
{and: [{x: 1}, {y: 2}], z: 3}
```

I am typing these examples from my head based on how I understand the code. I may be wrong.

@raymondfeng please take a second look at this change.

If the new implementation of `cast` is just an identity function (returning the input without any change), then I think we should mark `cast` as deprecated (at minimum in TSDocs; ideally also at runtime using [depd](https://www.npmjs.com/package/depd)) and remove it in the next major version.

This does not necessarily has to happen in this pull request, but it would be great to make those changes soon, otherwise we are increasing our technical debt.

Thoughts?